server/src/main/java/org/elasticsearch/plugins/PluginDescriptor.java

```diff
@@ -161,7 +166,16 @@ public void writeTo(final StreamOutput out) throws IOException {
         out.writeString(name);
         out.writeString(description);
         out.writeString(version);
-        Version.writeVersion(elasticsearchVersion, out);
+        if (out.getTransportVersion().before(TransportVersions.PLUGIN_DESCRIPTOR_STRING_VERSION)) {
+            try {
+                Version.writeVersion(Version.fromString(elasticsearchVersion), out);
+            } catch (IllegalArgumentException e) {
```

Comment:
If we can't parse the version, something is wrong. Let's not swallow the exception? But why would we ever expect this since we've successfully created this descriptor already? Seems like something we could validate earlier rather than every time we write out the descriptor to a stream.

Reply:
My thought was that if we can't parse the version, then we're in a (serverless) environment where the version is an arbitrary string. I suppose we could add a "has semantic version" field to the plugin descriptor, determine it at construction time, and use that for these switches. What do you think?